ETH-Merge-Bots/Milestones-in-TTD/src/agent.ts

```diff
@@ -0,0 +1,122 @@
+import { BlockEvent, Finding, HandleBlock } from "forta-agent";
+import {
+  getNumberOfDays,
+  getAverageBlockTime,
+  getEstimatedNumberOfBlocksUntilMerge,
+  getAvgBlockDifficulty,
+} from "./utils";
+import { TERMINAL_TOTAL_DIFFICULTY, MILESTONES, ETH_BLOCK_DATA } from "./eth.config";
+import { createFinding, createFinalFinding } from "./findings";
+import BigNumber from "bignumber.js";
+
+let blockCounter: number = 0;
+let firstTimestamp: number = 0;
+
+const avgBlockDifficulties: BigNumber[] = [];
+
+let isEmitted: { [key: string]: any } = {
+  low: false,
+  medium: false,
+  high: false,
+  critical: {
+    5: false,
+    4: false,
+    3: false,
+    2: false,
+    1: false,
+  },
+};
+
+let isMerged: boolean = false;
+
+export const provideHandleBlock = (ttd: BigNumber, ethBlockData: any, blockCounter: number): HandleBlock => {
+  return async (blockEvent: BlockEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+
+    const totalDifficulty = new BigNumber(blockEvent.block.totalDifficulty);
+
+    if (totalDifficulty.lt(ttd)) {
+      blockCounter++;
```

Comment:
Should this be incremented after L57? Based on the fact that for the first block, `currentBlockTimestamp - firstTimestamp = 0` in the numerator and `blockCounter=1` in the denominator of the `avgBlockTime`

Reply:
yes, good catch 👍